src/Map/src/Map.php

```diff
@@ -29,19 +30,18 @@ public function __construct(
         /**
          * @var array<Marker>
          */
-        private array $markers = [],
+        private SplObjectStorage $markers = new SplObjectStorage(),
```

Comment:
This is a nice improvement, but I would like to see this logic (with methods `*fromArray()` and `*toArray()`) extracted from `Map` to an other class.

Something like:
- `abstract class Elements`, with methods like `add`, `remove`, `toArray` and `fromArray`
- `class Markers extends Elements`
- `class Polygons extends Elements`
- `class Polylines extends Elements`

Anyway, doing this change is a BC for users using `new Map(markers: [new Marker()]`:
We can change it to: 
```suggestion
        array $markers = [],
```

and add a new property `private Markers $markers`.

Reply:
A bit like what's done with QueryParameters and co  in HttpFoundation ? Big yes for me. If we use objects implementing array access, indeedd no change required!